test/MUXControlsTestApp/InfoBar.cs

```diff
@@ -0,0 +1,473 @@
+            }
+            else if (Severity == InfoBarSeverity.Informational)
+            {
+                VisualStateManager.GoToState(this, "Informational", false);
+            }
+            else if (Severity == InfoBarSeverity.Success)
+            {
+                VisualStateManager.GoToState(this, "Success", false);
+            }
+            else if (Severity == InfoBarSeverity.None)
+            {
+                VisualStateManager.GoToState(this, "None", false);
+            }
+            else
+            {
+                VisualStateManager.GoToState(this, "Default", false);
+            }
+        }
+
+
+        // Updates visibility of buttons
+        void UpdateButtonsState()
+        {
+            if (ShowCloseButton)
+            {
+                if (CloseButtonContent != null && ActionButtonContent != null)
+                {
+                    VisualStateManager.GoToState(this, "BothButtonsVisible", false);
+                    VisualStateManager.GoToState(this, "NoDefaultCloseButton", false);
+                }
+                else if (CloseButtonContent != null)
+                {
+                    VisualStateManager.GoToState(this, "CloseButtonVisible", false);
+                    VisualStateManager.GoToState(this, "NoDefaultCloseButton", false);
+                }
+                else if (ActionButtonContent != null)
+                {
+                    VisualStateManager.GoToState(this, "ActionButtonVisible", false);
+                    VisualStateManager.GoToState(this, "DefaultCloseButton", false);
+                }
+                else if (ActionButtonContent == null && CloseButtonContent == null)
+                {
+                    VisualStateManager.GoToState(this, "NoButtonsVisible", false);
+                    VisualStateManager.GoToState(this, "DefaultCloseButton", false);
+                }
+            }
+            else
+            {
+                VisualStateManager.GoToState(this, "NoDefaultCloseButton", false);
+                if (ActionButtonContent != null)
+                {
+                    VisualStateManager.GoToState(this, "ActionButtonVisible", false);
+
+                }
+                else
+                {
+                    VisualStateManager.GoToState(this, "NoButtonsVisible", false);
+                }
+            }
+        }
+
+
+        // Updates if InfoBar is opened
+        void OnIsOpenChanged()
+        {
+            if (IsOpen)
+            {
+                lastCloseReason = InfoBarCloseReason.Programattic;
+
+            }
+            else
+            {
+                RaiseClosingEvent();
+                RaiseClosedEvent();
+            }
+            Open(IsOpen);
+        }
+
+
+        // Opens or closes the InfoBar
+        private void Open(bool value)
+        {
+            if (_myBorder != null)
+            {
+
+                if (value)
+                {
+                    _myBorder.Visibility = Visibility.Visible;
```

Comment:
Consider controlling this via a VisualState instead of programmatically.

Reply:
Yes, by having a visual state do this you could even remove the border as a required template part all together.

---
In reply to: [448016393](https://github.com/microsoft/microsoft-ui-xaml/pull/2788#discussion_r448016393) [](ancestors = 448016393)